google/cloud/security/common/gcp_api/_base_client.py

```diff
@@ -69,3 +71,31 @@ def _execute(self, request):
             upstream.
         """
         return request.execute()
+
+    def _build_paged_result(self, request, api_stub, rate_limiter):
+        """Execute results and page through the results.
+
+        Args:
+            request: GCP API client request object.
+            api_stub: The API stub used to build the request.
+            rate_limiter: An instance of RateLimiter to use.
+
+        Returns:
+            API response object.
+
+        Raises:
+            When the retry is exceeded, exception will be thrown.  This
+            exception is not wrapped by the retry library, and will be handled
+            upstream.
+        """
+        results = []
+        while request is not None:
+            try:
+                with rate_limiter:
+                    response = self._execute(request)
+                    results.append(response)
```

Comment:
QQ: previously, this was using ```.extend()```, which merged the page results into one list, rather than ```.append()``` which would create separate page results inside a list, IIRC.